6.analyze_features/scripts/9a.mean_aggregated_direct_hypothesis_test.py

```diff
@@ -0,0 +1,270 @@
+        "levene_p_value": "unsel_vs_wt_levene_p_value",
+        "holm_bonferroni_p_value": "unsel_vs_wt_levene_p_value_holm_bonferroni",
+    },
+    inplace=True,
+)
+high_vs_unsel.rename(
+    columns={
+        "levene_p_value": "high_vs_unsel_levene_p_value",
+        "holm_bonferroni_p_value": "high_vs_unsel_levene_p_value_holm_bonferroni",
+    },
+    inplace=True,
+)
+
+
+# drop columns
+high_vs_wt.drop(columns=["group", "levene_statistic"], inplace=True)
+unsel_vs_wt.drop(columns=["group", "levene_statistic"], inplace=True)
+high_vs_unsel.drop(columns=["group", "levene_statistic"], inplace=True)
+
+
+# combine the data
+combined_df = pd.merge(high_vs_wt, unsel_vs_wt, on="feature")
+combined_df = pd.merge(combined_df, high_vs_unsel, on="feature")
+combined_df.head()
+
+
+# In[4]:
+
+
+variance_df_path = pathlib.Path(
+    "../../data/6.analysis_results/mean_aggregated_variance_results_each_feature.csv"
+).resolve(strict=True)
+variance_df = pd.read_csv(variance_df_path)
+variance_df.head()
+
+
+# In[5]:
+
+
+# replace metadata genotype with the correct group
+variance_df["Metadata_genotype"].replace(
+    {"high": "High-Severity", "unsel": "Mid-Severity", "wt": "Wild Type"}, inplace=True
+)
+variance_df.head()
+
+
+# In[6]:
+
+
+high_severity = variance_df.loc[variance_df["Metadata_genotype"] == "High-Severity"]
+mid_severity = variance_df.loc[variance_df["Metadata_genotype"] == "Mid-Severity"]
+wt = variance_df.loc[variance_df["Metadata_genotype"] == "Wild Type"]
+
+# rename the columns
+high_severity.rename(columns={"variance": "high_severity_variance"}, inplace=True)
+mid_severity.rename(columns={"variance": "mid_severity_variance"}, inplace=True)
+wt.rename(columns={"variance": "wild_type_severity_variance"}, inplace=True)
+
+# drop columns
+high_severity.drop(columns=["Metadata_genotype"], inplace=True)
+mid_severity.drop(columns=["Metadata_genotype"], inplace=True)
+wt.drop(columns=["Metadata_genotype"], inplace=True)
+
+high_severity.reset_index(drop=True, inplace=True)
+mid_severity.reset_index(drop=True, inplace=True)
+wt.reset_index(drop=True, inplace=True)
+
+
+# In[7]:
+
+
+all_severities = pd.merge(
+    mid_severity, high_severity, left_on="feature", right_on="feature"
+)
+all_severities = pd.merge(all_severities, wt, left_on="feature", right_on="feature")
+combined_df = pd.merge(combined_df, all_severities, on="feature")
+combined_df.head()
+
+
+# In[8]:
+
+
+# make a new column that contains permutation test results for each feature
+np.random.seed(0)
+n_permutations = 1000
+# permute each column
+permuted_df = combined_df.copy()
+for column in permuted_df.columns:
+    if column != "feature":
+        permuted_df[column] = np.random.permutation(permuted_df[column].values)
+
+
+# In[9]:
+
+
+combined_df.head()
+
+
+# In[31]:
+
+
+dict_of_dfs = {"non_permuted": combined_df, "permuted": permuted_df}
+output_dfs = {"non_permuted": "", "permuted": ""}
+for key in dict_of_dfs:
+    df = dict_of_dfs[key]
+    # variance logic
```

Comment:
```suggestion
    # We are testing the specific hypothesis based on variance, where we're finding relationships based on significant levene's tests
```

I don't like my suggestion, but I recommend considering being more specific in this documentation

Reply:
oh, but then based on my reading of the `levene logic` section, i see you're not basing it off of the levene's test, just the variance.

(clear variable naming and documentation will help with these sorts of misunderstandings)

Reply:
added a markdown cell for increased clarity!